src/MainWindow.vala

```diff
@@ -410,6 +421,18 @@ namespace Terminal {
                 term.tab.icon = null; // Assume only process icons are set
             });
 
+            notebook.tab_bar.notify["autohide"].connect (() => {
+                new_tab_revealer.reveal_child =
+                    notebook.tab_bar.autohide &&
+                    notebook.tab_view.n_pages == 1;
+            });
+
+            notebook.tab_view.notify["n-pages"].connect (() => {
+                new_tab_revealer.reveal_child =
+                    notebook.tab_bar.autohide &&
+                    notebook.tab_view.n_pages == 1;
+            });
+
```

Comment:
It shouldn't use `tab_bar.autohide`, it should rely on gsettings instead.

Reply:
"n-pages" is already exposed by TerminalView. You should use it instead of making tab_view public. But for this to work you need to explicitly notify when n-pages changes using `notify_property ("n-pages")`

Reply:
TabView is already directly accessed by MainWindow in many other places so making the suggested change would involve refactoring not directly related to the issue.   I agree that in principle it would be more "modular" to make both TabView and TabBar private members of TerminalView but would you be happy to defer that to a separate code improvement PR?  Probably best left until after the Gtk4 port now in order avoid delaying that?

Reply:
I'm fine with that